TDDWithURMAndKotlin/part3_jump.md

```diff
@@ -0,0 +1,176 @@
+
+    jump(registry, 5, 10, instructionSet, 1)
+
+    assertEquals(1, instructionSet.current)
+}
+```
+
+Wow, that's actually an ugly function. It takes 5 parameters, probably Uncle Bob is looking at us and feeling disappointed… well, we will refactor it later. For now let's see if it works… well, actually we know it doesn't work, as [we see previously](https://dev.to/sierisimo/tdd-with-urm-and-kotlin-1dj7), if we run `gradle check` to run the tests, it will generate a _compile-time error_ because we haven't created the `jump` function yet.
+
+Let's fix that:
+
+```kotlin
+fun jump(registry: Registry, positionX: Int, positionY: Int, instructionSet: ?, instruction: Int) {
+
+}
+```
+
+Before moving on, we need to define how our `instrcutionSet` will behave or what type of data will be. Things to keep in mind:
+
+- It should hold as much instructions as need it
+- Needs to know which is the current instruction as which one is the next one based on the index
+- Needs to easily change the pointer of execution to any other instruction
+
+We could need more functionality from the `instructionSet` but having this 3 requirements tell us we need a class and will let us at least write the minimum necessary to fullfill our failing test. First we need to change the test adding a type for `instructionSet`:
+
+```kotlin
+internal class OperationsTest {
+    //…
+
+    val instructionSet = InstructionSet()
+}
+```
+
+And we can use it in the test, but wait, the test requires the `instructionSet` to have a `current` property. Let's create the class and add the property to remove the _compile-time error_:
+
+```kotlin
+class InstructionSet {
+    var current: Int = 0
+}
+```
+
+We use `var` because we will be setting the value to a different number on every execution of the instruction set. This class just holds data and actually doesn't need to do anything else for our current porpouse. Our test should pass now right? No compile time errors and it's quite straight forward.
+
+Well, not exactly. If we run the classic: `gradle check` we will get:
+
+```none
+expected: <1> but was: <0>
+org.opentest4j.AssertionFailedError: expected: <1> but was: <0>
+...
+```
+
+Hmmm we forgot to add an implementation to the function! Duh! Well… time to write the actual code:
+
+```kotlin
+fun jump(registry: Registry, positionX: Int, positionY: Int, instructionSet: InstructionSet, instruction: Int) {
+    instructionSet.current = instruction
+}
+```
+
+Easy peasy, now the test is passing! You can check by yourself. Please [go on this commit, which contains all of this code to this part and check it by yourself](https://github.com/sierisimo/publications/commit/929e3bb3dfff455fed5dccb51a2f716e5e43ebeb).
+
+At this point you probably are thinking I'm crazy, but I'm not. This is a fair thing happening on TDD, you created a test, you create code to pass the test, let's move on. No one said the implementation is right but the test is passing. We need to test deeper
```

Comment:
```suggestion
At this point you are probably thinking I'm crazy, but I'm not. This is a fair thing happening on TDD: you created a test, then you created code to pass the test, let's move on. No one said the implementation is right but the test is passing. We need to test deeper:
```

Reply:
Consistency of tenses, just to keep the tension going (I won't stop with the puns)